### PR TITLE
on failure only nack/ack message after sending to next instance

### PR DESCRIPTION
**Pull request by madmuffin1**

### CODE REVIEW

**Review by lrotermund — APPROVED** [review]

File: middleware/errorcounter.go [hunk 1]
```diff
@@ -30,23 +36,35 @@ func ErrorCounterWithConfig(config ErrorCounterConfig) gorabbitmq.MiddlewareFunc
...
 				if table == nil {
 					table = make(map[string]interface{})
 				}
 
 				if errorCounter, ok := c.Value(ErrorCounter{}).(int32); ok {
 					if errorCounter >= config.MaxRetries {
-						config.MaxRetriesExceeded(c)
+						retryError := config.MaxRetriesExceeded(c)
+						if retryError != nil {
+							return errors.Wrap(retryError, fmt.Sprintf("failed to execute maxRetriesExceeded handler, after %v", err))
```
Comment (Wr4thon) [inline middleware/errorcounter.go:48]: Please use the `Wrapf` function, provided by the `"github.com/pkg/errors"` package, so we dont have to import the `"fmt"` package.

File: middleware/errorcounter.go [hunk 2]
```diff
@@ -30,23 +36,35 @@ func ErrorCounterWithConfig(config ErrorCounterConfig) gorabbitmq.MiddlewareFunc
...
 					}
 
 					table[errorCounterKey] = errorCounter + 1
 				}
 
-				c.Queue().SendWithTable(c.DeliveryContext(), c.Delivery().Body, table)
+				// requeue message
+				requeueErr := c.Queue().SendWithTable(c.DeliveryContext(), c.Delivery().Body, table)
+				if requeueErr != nil {
+					return errors.Wrap(requeueErr, fmt.Sprintf("failed to requeue message, after %v", err))
```
Comment (Wr4thon) [inline middleware/errorcounter.go:62]: Please use the `Wrapf` function, provided by the `"github.com/pkg/errors"` package, so we dont have to import the `"fmt"` package.

File: middleware/errorcounter.go (hunk 1 above)
Comment (madmuffin1) [reply middleware/errorcounter.go:48]: fixed with 6fb874b08be0e80f9026fc22c75d7c7b0503be72

File: middleware/errorcounter.go (hunk 2 above)
Comment (madmuffin1) [reply middleware/errorcounter.go:62]: fixed with 6fb874b08be0e80f9026fc22c75d7c7b0503be72

**Pull request merged**